universum/configuration_support.py

```diff
@@ -8,14 +8,133 @@
+        should be the arguments, added to the command (e.g. ``--debug`` or ``-la``). Every command line element,
+        separated with space character, should be passed as a separate string argument. Lists like ``["ls -a"]``
+        will not be processed correctly and therefore should be splat into ``["ls", "-a"]``. Lists like
+        ``["build.sh", "--platform A"]`` will not be processed correctly and thus should be plat into
+        ``["build.sh", "--platform", "A"]``. A build step can have an empty list as a command - such step won't do
+        anything except showing up the step name in Universum execution logs. Some common actions, such as ``echo``
+        or ``cp``, are bash features and not actual programs to run. These features should be called as
+        ``["bash", "-c", "echo -e 'Some line goes here'"]``. Note that in this case the string to be passed to bash
+        is one argument containing white spaces and therefore not splat by commas.
+    environment
+        Required environment variables, e.g. ``environment={"VAR1": "String", "VAR2": "123"}`` Can be set at any
+        step level, but re-declaring variables is not supported, so please make sure to mention every variable only
+        one time at most.
+    artifacts
+        Path to the file or directory to be copied to the working directory as an execution
+        result. Can contain shell-style pattern matching (e.g. `"out/*.html"`), including recursive wildcards
+        (e.g. `"out/**/index.html"`). If not stated otherwise (see ``--no-archive``
+        `command-line parameter <args.html#Artifact\\ collection>`__ for details), artifact directories are copied
+        as archives. If :ref:`'artifact_prebuild_clean' key <clean_artifacts>` is either absent or set to `False`
+        and stated artifacts are present in downloaded sources, it is considered a failure and configuration
+        execution will not proceed. If no required artifacts were found in the end of the `Universum` run, it is
+        also considered a failure. In case of shell-style patterns build is failed if no files or directories
+        matching pattern are found.
+    report_artifacts
+        Path to the special artifacts for reporting (e.g. to Swarm). Unlike
+        :ref:`artifacts <build_artifacts>`, `report_artifacts` are not obligatory and their absence is not
+        considered a build failure. A directory cannot be stored as a separate artifact, so when using
+        ``--no-archive`` option, do not claim directories as `report_artifacts`. Please note that any file can be
+        put as `artifact`, `report_artifact`, or both. A file that is both in `artifacts` and `report_artifacts`,
+        will be mentioned in a report and will cause build failure when missing.
+    artifact_prebuild_clean
+        A flag to signal that directory should be clean prior to execution. By default,
```

Comment:
This flag signals that steps-specific artifacts must be cleaned before the build. This could be a single file, several files, shell-style pattern or directory.